drivers/timer/cortex_m_systick.c

```diff
@@ -115,6 +119,13 @@ static uint32_t elapsed(void)
 	 * So the count in val2 is post-wrap and last_load needs to be
 	 * added if and only if COUNTFLAG is set or val1 < val2.
 	 */
+	if (val1 == 0) {
+		val1 = last_load;
+	}
+	if (val2 == 0) {
+		val2 = last_load;
+	}
```

Comment:
I guess I'm having trouble understanding the logic here?  Seems like this is going to implement a N-1 cycle wraparound, which will lose time?  Or maybe that's OK with systick as the hardware is racy anyway.

I guess my gut says that what you want to do here if you detect you're in the 1-cycle race is just to retry the operation in the "safe" zone, no?

Reply:
No cycles are lost. The naming of `last_load` is slightly confusing as it is the full N cycles, the timer is actually (re)loaded with `last_load - 1` cycles.

> I guess my gut says that what you want to do here if you detect you're in the 1-cycle race is just to retry the operation in the "safe" zone, no?

That would be another option, but it seems unnecessary given this simple fix